servicetalk-concurrent-api/src/main/java/io/servicetalk/concurrent/api/DefaultExecutor.java

```diff
@@ -157,7 +158,7 @@ private static void shutdownExecutor(java.util.concurrent.Executor jdkExecutor)
             try {
                 ((AutoCloseable) jdkExecutor).close();
             } catch (Exception e) {
-                throw new RuntimeException("unexpected exception while closing executor: " + jdkExecutor, e);
+                throwException(e);
```

Comment:
most cases I added an exception, but some cases were either expected to be rare or otherwise seemed to make sense to re-throw or choose a use some other unchecked exception

Reply:
This will potentially change the type of the exception thrown. Wrapping the exception as a `RuntimeException` is gross, but it is at least consistent.

Reply:
both ways are not ideal, don't have any strong preference

Reply:
I will revert the change for this PR and submit in a followup for discussion and to make it more clear in release notes if we pull it in.

Reply:
moved to https://github.com/apple/servicetalk/pull/2228